bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/gtk/org/eclipse/swt/dnd/ClipboardProxy.java

```diff
@@ -169,6 +169,9 @@ boolean setData(Clipboard owner, Object[] data, Transfer[] dataTypes, int clipbo
 				System.arraycopy(entries, 0, tmp, 0, entries.length);
 				tmp[entries.length] = entry;
 				entries = tmp;
+				TransferData tdata = new TransferData();
```

Comment:
This change is non-functional. Why it was added?

Reply:
These 3 lines where needed to get Eclipse to present text/plain;uft-8 data to the clipboard.
Otherwise the javaToNative function in Texttreanfer.java wasebt called.

Reply:
This I confirmed by the eclipse debugger

Reply:
This is wrong - have you asked why does the rest of clipboard data works but only new type would need that? I would assume you've missed something else.

The javaToNative is called from `org.eclipse.swt.dnd.ClipboardProxy.getFunc(long, long, long, long)`